install/cupy_builder/_command.py

```diff
@@ -202,7 +202,7 @@ def build_extension(self, ext: setuptools.Extension) -> None:
         # Remove device code from list of sources, and instead add compiled
         # object files to link.
         ext.sources = sources_cpp
-        ext.extra_objects += extra_objects
+        ext.extra_objects = extra_objects + ext.extra_objects
```

Comment:
Note: The order of the command line matters: `libcudart_static.a` must be after `cupy_cub.cu.o` (which requires symbols in CUDA Runtime).